pvlib/iotools/tmy.py

```diff
@@ -198,9 +214,16 @@ def read_tmy3(filename, coerce_year=None, recolumn=True):
     # NOTE: as of pvlib-0.6.3, min req is pandas-0.18.1, so pd.to_timedelta
     # unit must be in (D,h,m,s,ms,us,ns), but pandas>=0.24 allows unit='hour'
     data.index = data_ymd + pd.to_timedelta(shifted_hour, unit='h')
-
-    if recolumn:
+    if map_variables is None:
         data = _recolumn(data)  # rename to standard column names
+        warnings.warn(
+            'TMY3 variable names will be renamed to pvlib conventions by '
+            'default starting in pvlib 0.9.5. Specify map_variables=True '
+            'to enable that behavior now, or specify map_variables=False '
+            'to hide this warning.', pvlibDeprecationWarning)
+        map_variables = False
+    if map_variables:
+        data = _recolumn(data).rename(columns=VARIABLE_MAP)
```

Comment:
I think that it would be nice to completely get rid of the ``_recolumn()`` function and do the renaming directly from the raw names.

```suggestion
    if map_variables:
        data = data.rename(columns=VARIABLE_MAP)
```